aiosql/query_loader.py

```diff
@@ -17,10 +18,20 @@ def __init__(self, driver_adapter: DriverAdapterProtocol, record_classes: Option
         self.driver_adapter = driver_adapter
         self.record_classes = record_classes if record_classes is not None else {}
 
-    def _make_query_datum(self, query_str: str, ns_parts: List):
+    def _make_query_datum(self, query_str: str, ns_parts: List) -> QueryDatum:
         lines = [line.strip() for line in query_str.strip().splitlines()]
-        query_name = lines[0].replace("-", "_")
+        operation_type, query_name = self._extract_operation_type(lines[0])
```

Comment:
Thanks for breaking this up into separate functions. Reads better, probably easier to maintain and change. :sparkles: